rpcs3/Emu/System.h

```diff
@@ -16,6 +16,18 @@ enum class system_state
 	ready,
 };
 
+enum game_boot_result : u32
```

Comment:
```suggestion
enum class game_boot_result : u32
```

Reply:
I'd rather not, class enums can't be implictly converted to booleans which means rewriting all if statements in this PR.

Reply:
then why not overloading operator bool for it?

Reply:
```constexpr operator bool(game_boot_result res)
{
    return res != game_boot_result{};
}
```

Reply:
Sadly not supported for class enums

Reply:
Could sacrifice lack of operator bool for enum class I think.

Reply:
On second though yeah, seems more sensible.